src/main/cpp/src/cast_string.cu

```diff
@@ -176,7 +584,59 @@ struct row_valid_fn {
   bitmask_type const* string_col_null_mask;
 };
 
+/**
+ * @brief validates a column has no ansi errors, throws if error
+ *
+ * @param col column to verify
+ * @param source_col original string column used to create col
+ * @param stream stream on which to operate
+ */
+void validate_ansi_column(column_view const& col,
+                          strings_column_view const& source_col,
+                          rmm::cuda_stream_view stream)
+{
+  auto const num_nulls      = col.null_count();
+  auto const incoming_nulls = source_col.null_count();
+  auto const num_errors     = num_nulls - incoming_nulls;
+  if (num_errors > 0) {
+    auto const first_error = thrust::find_if(rmm::exec_policy(stream),
+                                             thrust::make_counting_iterator(0),
+                                             thrust::make_counting_iterator(col.size()),
+                                             row_valid_fn{col.null_mask(), source_col.null_mask()});
+
+    offset_type string_bounds[2];
+    cudaMemcpyAsync(&string_bounds,
+                    &source_col.offsets().data<offset_type>()[*first_error],
+                    sizeof(offset_type) * 2,
+                    cudaMemcpyDeviceToHost,
+                    stream.value());
+    stream.synchronize();
```

Comment:
If we need copy then sync stream then why not just use `cudaMemcpy` (sync copy)?